The finally block always gets executed even when the return statement is
written at the end of the try block and the catch block. It always
executes , whether there is an exception or not. There are only a few
situations in which the finally block does not execute, such as VM
crash, power failure, software crash, etc. If you don't want to execute
the finally block, you need to call the System.exit() method explicitly
in the finally block.